src/ui/ai/AiRound.ts

```diff
@@ -225,9 +225,10 @@ export default class AiRound implements AiRoundInterface, PromotingInterface {
   public onEngineMove = (bestmove: string) => {
     const from = <Key>bestmove.slice(0, 2)
     const to = <Key>bestmove.slice(2, 4)
+    const role = <Role>chessFormat.uciToProm(bestmove)
```

Comment:
No need to cast here, in fact it's technically incorrect (could be `undefined` if the UCI isn't a promo move, which is OK for the `addMove` call)
```suggestion
    const role = chessFormat.uciToProm(bestmove)
```

Reply:
It's in fact not obvious to me 🙂  Happy to help.